### PR TITLE
Never show the enum backing field in the completion list

### PR DESCRIPTION
**Pull request by rchande**
Also, add a test for #4136.

Reviewers: @balajikris @basoundr @brettfo @dpoeschl @DustinCampbell @jasonmalinowski @Pilchie

### CODE REVIEW

**Comment by Pilchie** [comment]
LGTM.

**Comment by DustinCampbell** [comment]
:+1:

File: src/Workspaces/VisualBasic/Portable/Recommendations/VisualBasicRecommendationService.vb [hunk 1]
```diff
@@ -189,6 +189,7 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Recommendations
             Dim excludeShared = True ' do not show shared members by default
             Dim useBaseReferenceAccessibility = False
             Dim inNameOfExpression = node.IsParentKind(SyntaxKind.NameOfExpression)
+            Dim filterToEnumMembers = False
```
Comment (dpoeschl) [inline src/Workspaces/VisualBasic/Portable/Recommendations/VisualBasicRecommendationService.vb]: unused?

Comment (rchande) [inline src/Workspaces/VisualBasic/Portable/Recommendations/VisualBasicRecommendationService.vb]: Saving files is hard, will remove.

File: src/EditorFeatures/VisualBasicTest/Completion/CompletionProviders/SymbolCompletionProviderTests.vb [hunk 2]
```diff
@@ -6536,6 +6536,44 @@ End Class
             VerifyItemExists(text, "y")
         End Sub
 
+        <WorkItem(4136, "https://github.com/dotnet/roslyn/issues/4136")>
+        <Fact, Trait(Traits.Feature, Traits.Features.Completion)>
+        Public Sub No__ValueWhenDottingIntoEnum()
```
Comment (dpoeschl) [inline src/EditorFeatures/VisualBasicTest/Completion/CompletionProviders/SymbolCompletionProviderTests.vb]: Nit: Test name implies the field is named "__value" when it's actually named "value__" (I couldn't remember which it was and the mismatch between the method name and the ```VerifyItemIsAbsent``` was confusing).

**Comment by dpoeschl** [comment]
:+1:

**Pull request merged**